### PR TITLE
Delete unused installer properties

### PR DESCRIPTION
**Pull request by ViktorHofer**
In response to https://github.com/dotnet/runtime/issues/48338#issuecomment-782797482. Did an org search on every of these properties, they are all dead.

cc @am11

### CODE REVIEW

File: src/installer/Directory.Build.targets [hunk 1]
```diff
@@ -82,55 +72,6 @@
       <UseShippedHostPolicyPackage>false</UseShippedHostPolicyPackage>
```
Comment (am11) [inline src/installer/Directory.Build.targets:72]: For example, this is hardcoded to `false` (we can't control it from anywhere), so I was thinking maybe `HostPolicyVersion` which is falling back to 2.0.0 can be safely removed? Even if we want to keep `HostPolicyVersion` etc. for some reason, the hardcoded `false` values make these additional switch properties useless. Only way is to update this file manually, so we can leave line 73 unconditional and delete 71 and 74 (same for other similar props above).

**Comment by am11** [comment]
Great much easier to follow now! 🎉

File: src/installer/Directory.Build.targets (hunk 1 above)
Comment (ViktorHofer) [reply src/installer/Directory.Build.targets:72]: Thanks for the suggestion. I just took a look and I think the properties make sense the way they currently are (based on the descriptions above each block). Hopefully we can move these properties over to src\native\corehost\ after consolidating building and packaging of the host.

Comment (am11) [reply src/installer/Directory.Build.targets:72]: Yup, also `UseShippedHostPolicyPackage` etc. are only used in this file. I think if we get some feedback whether we solely rely on git branching and always build with same versions, these individual version properties can be unified in eng/Versions.props. Then we won't be needing this target.

File: src/installer/Directory.Build.targets [hunk 2]
```diff
@@ -39,11 +32,8 @@
       <ProductVersion>$(ProductionVersion)$(ProductVersionSuffix)</ProductVersion>
 
       <SharedFrameworkNugetVersion>$(ProductVersion)</SharedFrameworkNugetVersion>
-
-      <RuntimeDepsRpmVersion Condition="'$(RuntimeDepsRpmVersion)'==''">$(ProductVersion)</RuntimeDepsRpmVersion>
-      <RuntimeDepsDebVersion Condition="'$(RuntimeDepsDebVersion)'==''">$(ProductVersion)</RuntimeDepsDebVersion>
-
       <NuGetVersion>$(SharedFrameworkNugetVersion)</NuGetVersion>
+      <InstallersRelativePath>Runtime/$(SharedFrameworkNugetVersion)/</InstallersRelativePath>
```
Comment (am11) [inline src/installer/Directory.Build.targets:36]: nit: could inline the value at four call-sites at https://github.com/dotnet/runtime/blob/e5c7b036a094a420040c706b4fd275c2cf153a09/src/installer/publish/prepare-artifacts.proj#L98

Comment (ViktorHofer) [reply src/installer/Directory.Build.targets:36]: I hope that a big chunk of the publishing infra under installer can be removed in favor of Arcade's publishing infra. I need to create an issue for that to start the discussion (and understand the constraints). It's unclear to me why we are still doing so much of the publishing ourselves.

**Review by adamsitnik — APPROVED** [review]

**Pull request merged**